iot/http_example/cloudiot_http_example_nodejs.js

```diff
@@ -0,0 +1,155 @@
+        description: 'The Project ID to use. Defaults to the value of the GCLOUD_PROJECT or GOOGLE_CLOUD_PROJECT environment variables.',
+        requiresArg: true,
+        type: 'string'
+      },
+      cloud_region: {
+        default: 'us-central1',
+        description: 'GCP cloud region.',
+        requiresArg: true,
+        type: 'string'
+      },
+      registry_id: {
+        description: 'Cloud IoT registry ID.',
+        requiresArg: true,
+        demandOption: true,
+        type: 'string'
+      },
+      device_id: {
+        description: 'Cloud IoT device ID.',
+        requiresArg: true,
+        demandOption: true,
+        type: 'string'
+      },
+      private_key_file: {
+        description: 'Path to private key file.',
+        requiresArg: true,
+        demandOption: true,
+        type: 'string'
+      },
+      algorithm: {
+        description: 'Encryption algorithm to generate the JWT.',
+        requiresArg: true,
+        demandOption: true,
+        choices: ['RS256', 'ES256'],
+        type: 'string'
+      },
+      num_messages: {
+        default: 100,
+        description: 'Number of messages to publish.',
+        requiresArg: true,
+        type: 'number'
+      },
+      http_bridge_address: {
+        default: 'cloudiot-device.googleapis.com',
+        description: 'HTTP bridge address.',
+        requiresArg: true,
+        type: 'string'
+      },
+      message_type: {
+        default: 'events',
+        description: 'Message type to publish.',
+        requiresArg: true,
+        choices: ['events', 'state'],
+        type: 'string'
+      }
+    })
+    .example(`node $0 cloudiot_http_example_nodejs.js --project_id=blue-jet-123 --registry_id=my-registry --device_id=my-node-device --private_key_file=../rsa_private.pem --algorithm=RS256`)
+    .wrap(120)
+    .recommendCommands()
+    .epilogue(`For more information, see https://cloud.google.com/iot-core/docs`)
+    .help()
+    .strict()
+    .argv;
+
+// Create a Cloud IoT Core JWT for the given project ID, signed with the given
+// private key.
+function createJwt (projectId, privateKeyFile, algorithm) {
+  // Create a JWT to authenticate this device. The device will be disconnected
+  // after the token expires, and will have to reconnect with a new token. The
+  // audience field should always be set to the GCP project ID.
+  const token = {
+    'iat': parseInt(Date.now() / 1000),
+    'exp': parseInt(Date.now() / 1000) + 20 * 60,  // 20 minutes
+    'aud': projectId
+  };
+  const privateKey = fs.readFileSync(privateKeyFile);
+  return jwt.sign(token, privateKey, { algorithm: algorithm });
+}
+
+// Publish numMessages message asynchronously, starting from message
+// messageCount. Telemetry events are published at a rate of 1 per second and
+// states at a rate of 1 every 2 seconds.
+function publishAsync (messageCount, numMessages) {
+  const payload = `${argv.registry_id}/${argv.device_id}-payload-${messageCount}`;
```

Comment:
The `publishAsync` function in `http_client_example.js` seems more flexible allowing the user to specify a `message` argument.  Why is the `publishAsync` function in this file not allowing the same?

Reply:
Checked in duplicate versions of the same sample, will clean up after dinner.